src/System.Memory/src/System/Buffers/ReadOnlySequence.cs

```diff
@@ -128,20 +125,48 @@ public ReadOnlySequence(T[] array, int start, int length)
         /// </summary>
         public ReadOnlySequence(ReadOnlyMemory<T> readOnlyMemory)
         {
-            ReadOnlySequenceSegment segment = new ReadOnlySequenceSegment
+            if (MemoryMarshal.TryGetOwnedMemory(readOnlyMemory, out OwnedMemory<T> ownedMemory))
```

Comment:
Add `ReadOnlySequenceFactory` implementations and test class variations for these cases.

Reply:
Done